quix-frontend/client/src/lib/sql-autocomplete/adapter/sql-autocpmplete-adapter-utills.ts

```diff
@@ -0,0 +1,304 @@
+}
+
+function startsWithRow(trinoObjectAsString: string, counter: number) {
+  return trinoObjectAsString.substring(counter, counter + 4) === StartingOptions.row;
+}
+
+function getOperator(trinoObjectAsString: string, start: number , end: number) {
+  return trinoObjectAsString.substring(start+1, end+1); //we are in middle of string start is ' ' 
+}
+
+function findClosingParentheses(trinoObjectAsString: string, counter: number): number { //rename
+  let parenthesis = 0;
+  for (let i = counter; i < trinoObjectAsString.length; i++) {
+    if (trinoObjectAsString.charAt(i) === '(') {
+      parenthesis++;
+    }
+    if (trinoObjectAsString.charAt(i) === ')') {
+      parenthesis--;
+    }
+    if (parenthesis === 0) {
+      return i
+    }
+  }
+  return -1;
+}
+
+function handleStartingOptions(trinoObjectAsString: string, start: number, end: number) {
+  let objectToInsert;
+  let newIndex = end;
+
+  switch (getOperator(trinoObjectAsString, start, end)) {
+    case StartingOptions.row: {
+      objectToInsert = trinoToJs(trinoObjectAsString, end - 3);
+      newIndex = findClosingParentheses(trinoObjectAsString, end);
+      break;
+    }
+    case StartingOptions.map: {
+      newIndex = findClosingParentheses(trinoObjectAsString, end);
+      return { objectToInsert, type: "map", newIndex };
+    }
+    case StartingOptions.array: {
+      newIndex = findClosingParentheses(trinoObjectAsString, end);
+      return { objectToInsert, type: "array", newIndex };
+    }
+    case StartingOptions.timestamp:  {
+      newIndex = findClosingParentheses(trinoObjectAsString, end);
+      return { objectToInsert, type: "timeStamp", newIndex };
+    }
+    default : {
+      throw new Error(`Error at index: ${trinoObjectAsString.substring(end-5,end)}`);
+    }
+  }
+
+  return { objectToInsert, newIndex };
+}
+
+function checkKey(key: string,indexInOriginalString:number) {
+  const invalidCharsInKey = ['!', '@', '#', '$', '%', '^', '&', '*', '(', ')', ',', '.', ':', ';', ' ', '\t', '\n'];
+  key.split('').forEach((char) => {
+    if (invalidCharsInKey.includes(char)) {
+      throw new Error(`Error at index: ${indexInOriginalString}, illegal key value ${char}`);
+    }
+  });
+}
+
+
+export function findRelevantPartOfPrefix(tables: any, brokenPrefix: string[]): string {
+  let relevantPartOfPrefix = '';
+
+  for (const table of tables) {
+      for (const column of table.columns) {
+          let found = false;
+          let gotRelevantPartOfPrefix = false;
+
+          brokenPrefix.forEach(cell => {
+              if (found) {
+                  relevantPartOfPrefix += cell + '.';
+              }
+              if (!gotRelevantPartOfPrefix && cell === column.name) {
+                  found = true;
+                  gotRelevantPartOfPrefix = true;
+                  relevantPartOfPrefix += cell + '.';
+              }
+          });
+          if (relevantPartOfPrefix) {
+            break;
```

Comment:
break is not a good practice.
in that case you can `return relevanPartOfPrefix;`

Reply:
fixed